nmdc_runtime/site/ops.py

```diff
@@ -1043,6 +1045,48 @@ def site_code_mapping() -> dict:
         )
 
 
+@op(
+    required_resource_keys={"mongo"},
+    config_schema={
+        "source_ontology": str,
+        "output_directory": Field(Noneable(str), default_value=None, is_required=False),
+        "generate_reports": Field(bool, default_value=True, is_required=False),
+    },
+)
+def load_ontology(context: OpExecutionContext):
+    cfg = context.op_config
+    source_ontology = cfg["source_ontology"]
+    output_directory = cfg.get("output_directory")
+    generate_reports = cfg.get("generate_reports", True)
+
+    if output_directory is None:
+        output_directory = os.path.join(os.getcwd(), "ontology_reports")
+
+    # Redirect Python logging to Dagster context
+    handler = logging.Handler()
+    handler.emit = lambda record: context.log.info(record.getMessage())
+
+    # Get logger from ontology_load package
+    controller_logger = logging.getLogger("ontology_loader.ontology_load_controller")
+    controller_logger.setLevel(logging.INFO)
+    controller_logger.addHandler(handler)
+
+    context.log.info(f"Running Ontology Loader for ontology: {source_ontology}")
+    loader = OntologyLoaderController(
+        source_ontology=source_ontology,
+        output_directory=output_directory,
+        generate_reports=generate_reports,
+    )
+
+    try:
+        loader.run_ontology_loader()
+        context.log.info("Ontology load completed successfully!")
+    except Exception as e:
+        context.log.info(f"Error running ontology loader: {e}")
```

Comment:
Do we want to catch this exception here? Would it be better to let the job fail if this op fails if the loader fails?

Reply:
removed exception; let job fail without catching